libraries/holochain-service-client/src/main/java/org/holochain/androidserviceruntime/holochain_service_client/HolochainServiceClient.kt

```diff
@@ -49,42 +50,126 @@ class HolochainServiceClient(
+
+        deferred.await()
     }
 
     /// Enable an installed app
-    fun enableApp(installedAppId: String) {
-        this.mService!!.enableApp(installedAppId)
+    suspend fun enableApp(installedAppId: String): AppInfoFfiParcel {
+        Log.d(TAG, "enableApp")
+        val deferred = CompletableDeferred<AppInfoFfiParcel>()
+        var callbackBinder = object : IHolochainServiceCallbackStub() {
+            override fun enableApp(response: AppInfoFfiParcel) {
+                Log.d(TAG, "enableApp callback")
+                deferred.complete(response)
+            }
+        }
+        this.mService!!.enableApp(callbackBinder, installedAppId)
+        return deferred.await()
     }
 
     /// Disable an installed app
-    fun disableApp(installedAppId: String) {
-        this.mService!!.disableApp(installedAppId)
+    suspend fun disableApp(installedAppId: String) {
+        Log.d(TAG, "disableApp")
+        val deferred = CompletableDeferred<Unit>()
+        var callbackBinder = object : IHolochainServiceCallbackStub() {
+            override fun disableApp() {
+                Log.d(TAG, "disableApp callback")
+                deferred.complete(Unit)
+            }
+        }
+        this.mService!!.disableApp(callbackBinder, installedAppId)
+        deferred.await()
     }
 
     /// List installed happs in conductor
-    fun listApps(): MutableList<AppInfoFfiParcel> {
-        return this.mService!!.listApps()
+    suspend fun listApps(): List<AppInfoFfiParcel> {
+        Log.d(TAG, "listApps")
+
+        val deferred = CompletableDeferred<List<AppInfoFfiParcel>>()
+        var callbackBinder = object : IHolochainServiceCallbackStub() {
+            override fun listApps(response: List<AppInfoFfiParcel>) {
+                Log.d(TAG, "listApps callback")
+                deferred.complete(response)
+            }
+        }
+        this.mService!!.listApps(callbackBinder)
+
+        return deferred.await()
     }
 
     /// Get or create an app websocket with authentication token
-    fun ensureAppWebsocket(installedAppId: String): AppAuthFfiParcel {
-        return this.mService!!.ensureAppWebsocket(installedAppId)
+    suspend fun ensureAppWebsocket(installedAppId: String): AppAuthFfiParcel {
+        Log.d(TAG, "ensureAppWebsocket")
+
+        val deferred = CompletableDeferred<AppAuthFfiParcel>()
+        var callbackBinder = object : IHolochainServiceCallbackStub() {
+            override fun ensureAppWebsocket(response: AppAuthFfiParcel) {
+                Log.d(TAG, "ensureAppWebsocket callback")
+                deferred.complete(response)
+            }
+        }
+        this.mService!!.ensureAppWebsocket(callbackBinder, installedAppId)
+
+        return deferred.await()
     }
 
-    fun signZomeCall(args: ZomeCallUnsignedFfiParcel): ZomeCallFfiParcel {
-        return this.mService!!.signZomeCall(args);
+    /// Sign a zome call
+    suspend fun signZomeCall(args: ZomeCallUnsignedFfiParcel): ZomeCallFfiParcel {
```

Comment:
`suspend fun` - that doesn't read like you enjoyed this a whole lot

Reply:
ba dum tiss :drum: